azurerm/internal/services/hdinsight/hdinsight_kafka_cluster_resource.go

```diff
@@ -298,9 +335,10 @@ func resourceArmHDInsightKafkaClusterRead(d *schema.ResourceData, meta interface
 		}
 
 		kafkaRoles := hdInsightRoleDefinition{
-			HeadNodeDef:      hdInsightKafkaClusterHeadNodeDefinition,
-			WorkerNodeDef:    hdInsightKafkaClusterWorkerNodeDefinition,
-			ZookeeperNodeDef: hdInsightKafkaClusterZookeeperNodeDefinition,
+			HeadNodeDef:            hdInsightKafkaClusterHeadNodeDefinition,
+			WorkerNodeDef:          hdInsightKafkaClusterWorkerNodeDefinition,
+			ZookeeperNodeDef:       hdInsightKafkaClusterZookeeperNodeDefinition,
+			KafkaManagementNodeDef: &hdInsightKafkaClusterKafkaManagementNodeDefinition,
```

Comment:
This should do no harm (and I refer to the same way which has been done for the [`EdgeNodeDef`](https://github.com/terraform-providers/terraform-provider-azurerm/blob/cffbabbcdf733487b7e926ca0e0fc5d932b2a12c/azurerm/internal/services/hdinsight/hdinsight_ml_services_cluster_resource.go#L172)). Because when you do expand, it will further check whether user has specify the corresponding schema; when you do flatten, it will further find the corresponding property by name (in this case `kafkamanagementnode`) in the response body. In both cases, if the kafka management node is not enabled, it shall do nothing.